src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenHandlerBase.cs

```diff
@@ -196,7 +209,15 @@ protected AcquireTokenHandlerBase(RequestData requestData)
             }
         }
 
-        private void StoreResultExToCache(ref bool notifiedBeforeAccessCache)
+        public static async Task<List<string>> GetOrderedAliases(string host, bool validateAuthority, CallState callState)
+        {
+            var metadata = await InstanceDiscovery.GetMetadataEntry(host, validateAuthority, callState).ConfigureAwait(false);
+            var aliasedAuthorities = new List<string>(new string[] { metadata.PreferredCache, host });
+            aliasedAuthorities.AddRange(metadata.Aliases ?? Enumerable.Empty<string>());
+            return aliasedAuthorities;
+        }
+
+        private async Task<bool> StoreResultExToCache(bool notifiedBeforeAccessCache)
```

Comment:
why do we need to return bool here? we can simply update the global variable instead

Reply:
Because:

* That `notifiedBeforeAccessCache` thing is not global. It is a local variable in the calling function
* This method becomes an async method now, and an async method can not use the `ref` way to update the input variable
* Last but not the least, when a function is going to provide something back to the caller, it seems using return value would be the most intuitive way.